lambda/getSchedule/tias-app/Scheduler/src/main/java/db/Person.java

```diff
@@ -0,0 +1,208 @@
+        return email;
+    }
+
+    public String getFirstName() {
+        return firstName;
+    }
+
+    public String getLastName() {
+        return lastName;
+    }
+
+    public int getDesiredNumberAssignments() {
+        return desiredNumberAssignments;
+    }
+
+    public boolean isPeerTeacher() {
+        return peerTeacher;
+    }
+
+    public boolean isTeachingAssistant() {
+        return teachingAssistant;
+    }
+
+    public boolean isAdministrator() {
+        return administrator;
+    }
+
+    public boolean isProfessor() {
+        return professor;
+    }
+
+    public ArrayList<Availability> getAvailabilities() {
+        return (ArrayList<Availability>) Collections.unmodifiableList(availabilities);
+    }
+
+    public void addAvailability(Availability availability) {
+        availabilities.add(availability);
+    }
+
+    public ArrayList<Preference> getPreferences() {
+        return preferences;
+    }
+
+    public void addPreference(Preference preference) {
+        preferences.add(preference);
+    }
+
+    public void sortPreferences() {
+        Collections.sort(preferences, Collections.reverseOrder());
+    }
+
+    public ArrayList<Qualification> getQualifications() {
+        return (ArrayList<Qualification>) Collections.unmodifiableList(qualifications);
+    }
+
+    public void addQualification(Qualification qualification) {
+        qualifications.add(qualification);
+    }
+
+    public float getAvailabilityScore() {
+        return availabilityScore;
+    }
+
+    public ArrayList<Integer> getAvailabileSections() {
+        return (ArrayList<Integer>) Collections.unmodifiableList(availableSections);
+    }
+
+    public int getCurrentAssignments() {
+        return currentAssignments;
+    }
+    
+    public void addCurrentAssignment() {
+        currentAssignments++;
+    }
+
+    boolean isQualified(int courseId) {
+        for (Qualification qualification : qualifications) {
+            if (courseId == qualification.getCourseId()) {
+                return qualification.isQualified();
+            }
+        }
+        return false;
+    }
+
+    boolean isAvailable(ArrayList<Meeting> meetings) {
+        for (Meeting meeting : meetings) {
+            boolean quit = true;
+            for (Availability availability : availabilities) {
+                if (availability.getWeekday().equals(meeting.getWeekday()) && timeContains(availability.startTime, availability.endTime, meeting.startTime, meeting.endTime)) {
+                    quit = false;
+                }
+            }
+            if (quit) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    boolean getPreferenceEquals(int sectionId, DBPreference target) {
+        for (Preference preference : preferences) {
+            if (preference.getSectionId() == sectionId) {
+                return preference.getPreference() == target;
+            }
+        }
+        return false;
+    }
```

Comment:
**TODO**: Improve from *O(n)* to *O(1)* by converting `qualifications` from `ArrayList` to `HashSet`